api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -366,6 +396,10 @@ message Principal {
     // ``not_id`` would match, this principal would not match. Conversely, if the
     // value of ``not_id`` would not match, this principal would match.
     Principal not_id = 8;
+
+    // Matches against metadata from either dynamic state or route configuration. Preferred over the
+    // ``metadata`` field as it provides more flexibility in metadata source selection.
+    SourcedMetadata sourced_metadata = 13;
```

Comment:
I think you may need to deprecate the previsous `metadata` field. And I am not good at the naming, But the `SourcedMetadata` sounds a little weird.

Reply:
The intention was to convey that this would allow picking a `Source` for the metadata. Would `resolved_metadata` or `composite_metadata` be cleaner? If you have any other suggestions, I can change it to that.

Reply:
I also have no any idea. Compare to the new two name, seems sourced_metadata is fine. orz.